src/TextInputWithTokens/TextInputWithTokens.tsx

```diff
@@ -296,7 +297,7 @@ function TextInputWithTokensInnerComponent<TokenComponentType extends AnyReactCo
         visualPosition="leading"
         showLoadingIndicator={showLeadingLoadingIndicator}
       >
-        {typeof LeadingVisual === 'function' ? <LeadingVisual /> : LeadingVisual}
+        {typeof LeadingVisual !== 'string' && isValidElementType(LeadingVisual) ? <LeadingVisual /> : LeadingVisual}
```

Comment:
we have to check for `string` explicitly here, because things like `'div'` is a valid element type, but we don't want to try and render an empty div, but instead just pass through that text. 


We may want to consider whether accepting raw element types makes sense longterm (I don't think there's a good reason to accept a raw Component instead of just React.ReactNode).

```tsx
<Component leadingVisual={SearchIcon} /> // imo this shouldn't be allowed

<Component leadingVisual={<SearchIcon />} /> // imo this should be the way
```

but that might be a later breaking change at this point?

Reply:
Oh great point! Thanks for the note as well, that was one of my questions until I came to that point!

➕ 1  for using ReactNodes instead of raw components